libsplinter/src/rest_api/mod.rs

```diff
@@ -330,6 +330,93 @@ impl RequestGuard for Box<dyn RequestGuard> {
     }
 }
 
+/// Guards requests based on a minimum protocol version.
+///
+/// A protocol version is specified via the HTTP header `"X-SplinterProtocolVersion"`.  This header
+/// is a positive integer value.
+#[derive(Clone)]
+pub struct ProtocolVersionRangeGuard {
+    min: u32,
+    max: u32,
+}
+
+impl ProtocolVersionRangeGuard {
+    /// Constructs a new protocol version guard with the given minimum.
+    pub fn new(min: u32, max: u32) -> Self {
+        Self { min, max }
+    }
+}
+
+impl RequestGuard for ProtocolVersionRangeGuard {
+    fn evaluate(&self, req: &HttpRequest) -> Continuation {
+        if let Some(header_value) = req.headers().get("X-SplinterProtocolVersion") {
+            match header_value
+                .to_str()
+                .map_err(|err| {
+                    format!(
+                        "Invalid characters in X-SplinterProtocolVersion header: {}",
+                        err
+                    )
+                })
+                .and_then(|val_str| {
+                    val_str.parse::<u32>().map_err(|_| {
+                        format!("X-SplinterProtocolVersion must be a valid positive integer")
+                    })
+                }) {
```

Comment:
This is a long statement to match on; would be easier to read if declared as a variable first.

Reply:
Fixed